src/cargo/ops/cargo_new.rs

```diff
@@ -660,7 +660,15 @@ fn discover_author() -> CargoResult<(String, Option<String>)> {
         .or_else(|| get_environment_variable(&email_variables[3..]));
 
     let name = name.trim().to_string();
-    let email = email.map(|s| s.trim().to_string());
+    let email = email.map(|mut s| {
+        s = s.trim().to_string();
```

Comment:
This might be best done with a bit more slicing, like:

```rust
let mut s = s.trim();
if s.starts_with("<") && s.ends_with(">") {
    s = &s[1..s.len() - 1];
}
s.to_string()
```

Reply:
Also, could a comment be added why it's being trimmed?

Reply:
I'll fix it up with the slicing. But I actually do not know why there is a trim in there, it is from the original code.  My guess is that some sources might include a newline and we want to make sure that is gone.

Reply:
@alexcrichton PR has been updated with this change minus the comment.

Reply:
Ah yeah the `trim()` is sort of just for good measure, I'm not sure if it's solidly motivated one way or the other.

As a mild stylistic note as well, the `{` after the `if` clause typically comes on the same line instead of the next line

Reply:
Falling into my C habit.   Fixed.

Reply:
Ok thinks! And for a comment FWIW I was just thinking of your use case of stripping the angle brackets. Would you be ok adding that?

Reply:
Done.